main.go

```diff
@@ -456,177 +467,170 @@ func makeSecretChanges(kubeClient *k8s.Client, secret *corev1.Secret, initiator
 			return err
 		}
+		return nil
+	}
+	if err != nil {
+		return err
 	}
 
-	return nil
-}
+	// already exists
+	log.Info().Msgf("[%v] Secret %v.%v - Already exists in namespace %v, updating data...", initiator, secret.Name, secret.Namespace, namespace.Name)
 
-func isEventExist(kubeClient *k8s.Client, namespace string, name string, event *corev1.Event) (exist string, err error) {
-	err = kubeClient.Get(context.Background(), namespace, name, event)
+	// update data in secret
+	secretInNamespace.Data = secret.Data
+	secretInNamespace.Annotations[annotationLetsEncryptCertificateState] = secret.Annotations[annotationLetsEncryptCertificateState]
 
-	if apiErr, ok := err.(*k8s.APIError); ok {
-		if apiErr.Code == http.StatusNotFound {
-			return "not found", err
-		}
+	_, err = kubeClientset.CoreV1().Secrets(namespace.Name).Update(secretInNamespace)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
 
+func isEventExist(kubeClientset *kubernetes.Clientset, namespace string, name string) (*v1.Event, string, error) {
+	event, err := kubeClientset.CoreV1().Events(namespace).Get(name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		return nil, "not found", err
+	}
+	if err != nil {
 		log.Error().Msgf("Event %v.%v - Getting event has an error.\n\t%s", name, namespace, err.Error())
-		return "error", err
+		return nil, "error", err
 	}
 
-	return "found", nil
+	return event, "found", nil
 }
 
-func postEventAboutStatus(kubeClient *k8s.Client, secret *corev1.Secret, eventType string, action string, reason string, message string, kind string, reportingController string, reportingInstance string) (err error) {
+func postEventAboutStatus(kubeClientset *kubernetes.Clientset, secret *v1.Secret, eventType string, action string, reason string, message string, kind string, reportingController string, reportingInstance string) (err error) {
 	now := time.Now().UTC()
-	secs := int64(now.Unix())
 	count := int32(1)
-	eventName := fmt.Sprintf("%v-%v", *secret.Metadata.Name, action)
+	eventName := fmt.Sprintf("%v-%v", secret.Name, action)
 	eventSource := os.Getenv("HOSTNAME")
-	var eventResp corev1.Event
-	var exist string
-	exist, err = isEventExist(kubeClient, *secret.Metadata.Namespace, eventName, &eventResp)
+	eventResp, exist, err := isEventExist(kubeClientset, secret.Namespace, eventName)
 
 	if exist == "error" {
 		return err
 	}
 
 	if exist == "found" {
-		count = *eventResp.Count + 1
-		eventResp.Type = &eventType
-		eventResp.Action = &action
-		eventResp.Reason = &reason
-		eventResp.Message = &message
-		eventResp.Count = &count
-		eventResp.LastTimestamp.Seconds = &secs
-		err = kubeClient.Update(context.Background(), &eventResp)
+		count = eventResp.Count + 1
+		eventResp.Type = eventType
+		eventResp.Action = action
+		eventResp.Reason = reason
+		eventResp.Message = message
+		eventResp.Count = count
+		eventResp.LastTimestamp = metav1.NewTime(now)
+		_, err = kubeClientset.CoreV1().Events(secret.Namespace).Update(eventResp)
```

Comment:
Does the controller have event permissions on all namespaces?

Reply:
Yes https://github.com/estafette/estafette-letsencrypt-certificate/blob/master/helm/estafette-letsencrypt-certificate/templates/clusterrole.yaml#L29